src/app/pages/auth/services/auth.service.ts

```diff
@@ -27,34 +27,36 @@ export class AuthService {
     public http: HttpClient,
     private token: TokenService,
     private config: ConfigService,
-    private logger: LogService
+    private logger: LogService,
+    private analytics: AnalyticsService
   ) {}
 
   authenticate(authObj) {
     return (isValidURL(authObj)
-      ? this.URLAuth(authObj)
-      : this.nonURLAuth(authObj)
+      ? this.metaTokenUrlAuth(authObj)
+      : this.metaTokenJsonAuth(authObj)
     ).then(refreshToken => {
       return this.registerToken(refreshToken)
         .then(() => this.registerAsSource())
         .then(() => this.registerToken(refreshToken))
     })
   }
 
-  URLAuth(authObj) {
+  metaTokenUrlAuth(authObj) {
     // NOTE: Meta QR code and new QR code
     return this.getRefreshTokenFromUrl(authObj).then((body: any) => {
       this.logger.log(`Retrieved refresh token from ${body.baseUrl}`, body)
       const refreshToken = body.refreshToken
       return this.token
         .setURI(body.baseUrl)
+        .then(baseUrl => this.analytics.setUserProperties({ baseUrl }))
```

Comment:
I don't think this is needed anymore? Since you already set it [here](https://github.com/RADAR-base/RADAR-Questionnaire/blob/staging-app/src/app/pages/auth/services/auth.service.ts#L101), and the firebase user properties are set [here](https://github.com/RADAR-base/RADAR-Questionnaire/blob/staging-app/src/app/core/services/config/config.service.ts#L223).

Reply:
This is needed for login, if the auth client ID and secret depend on the installation:
1. scan meta QR code
2. update base URL, get new auth client secret from firebase
3. get subject information
4. update other Firebase user properties

If we omit step 2, step 3 fails with a "Bad Credentials" message.

Reply:
Oh okay, thanks for clarifying.